### PR TITLE
`{PannerNode,AudioListener}.{setPosition,setOrientation}` should throw when set while an automation curve is active.

### LINKED ISSUE #1268: Behaviour of the interaction between deprecated `setPosition` method on `PannerNode` and the new behaviour of the `AudioParam.value` setter
Consider the following:
- On a `PannerNode`, have one of the `AudioParam` be automated using a curve
- When the `AudioContext.currentTime` reaches the curve region, call `setPosition`

This is specced as being equivalent to setting the `AudioParam`s directly. This in turn is equivalent to calling `AudioParam.setValueAtTime(v, ac.currentTime)` with `v` the value being set and `ac` the `AudioContext`.

Should `setPosition` now throw ? Should it swallow the exception ?

### PR DESCRIPTION
**Pull request by padenot**
This fixes #1268.

***
[Preview](https://s3.amazonaws.com/pr-preview/padenot/web-audio-api/1268-throw-setPosition-and-al.html) | [Diff](https://s3.amazonaws.com/pr-preview/WebAudio/web-audio-api/ad70087...padenot:730940f.html)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -7614,7 +7614,8 @@ <h2 id="AudioWorklet-Examples">
             <p>
               This method is DEPRECATED. It is equivalent to setting
               <code>positionX</code>, <code>positionY</code>, and
-              <code>positionZ</code> AudioParams directly.
+              <code>positionZ</code> <a>AudioParam</a>s <code>value</code>
+              attribute directly.
```
Comment (rtoy) [inline index.html]: Not your fault, but that's pretty vague on what "setting" means.  If we clarify this to say the same as `positionX.setValueAtTime(x, context.currentTime)` and so on, would we need the rest of the text below?  I think if we say this, it makes it pretty clear what happens if there's an `setValueCurveAtTime` there.

**Comment by joeberkovitz** [comment]
@padenot Can you merge this if it's ready? It seems to be.

File: index.html (hunk 1 above)
Comment (padenot) [reply index.html]: I've decided to copy some prose I found for setOrientation, and make all that extra clear.

**Comment by padenot** [comment]
This has been rewrote to apply cleanly after the contiguous IDL changes, and it's also a bit clearer.

File: index.html [hunk 2]
```diff
@@ -10630,8 +10630,19 @@ <h2 id="AudioWorklet-Examples">
...
+                <code>orientationX.value</code>,
+                <code>orientationY.value</code>, and
+                <code>orientationZ.value</code> attribute directly, with the
+                <code>x</code>, <code>y</code> and <code>z</code> parameters,
+                respectively.
+              </p>
+              <p>
+                Consequently, if any of the <code>orientationX</code>,
+                <code>orientationY</code>, and <code>orientationZ</code>
+                <a>AudioParam</a> have an automation curve set using <a href=
```
Comment (rtoy) [inline index.html]: Typo: AudioParam -> AudioParams

File: index.html [hunk 3]
```diff
@@ -10630,8 +10630,19 @@ <h2 id="AudioWorklet-Examples">
...
+                "#dom-audioparam-setvaluecurveattime">setValueCurveAtTime()</a>
```
Comment (rtoy) [inline index.html]: I think the link should be `<a data-link-for="AudioParam">setValueCurveAtTime()</a>`.  Maybe parens outside the tag?

File: index.html [hunk 4]
```diff
@@ -10630,8 +10630,19 @@ <h2 id="AudioWorklet-Examples">
             <dd>
               <p>
                 This method is DEPRECATED. It is equivalent to setting
-                <code>orientationX</code>, <code>orientationY</code>, and
-                <code>orientationZ</code> AudioParams directly.
+                <code>orientationX.value</code>,
+                <code>orientationY.value</code>, and
+                <code>orientationZ.value</code> attribute directly, with the
```
Comment (rtoy) [inline index.html]: Should orientationX,Y,Z be linkified? Just asking; I'm ok if you don't do this.

File: index.html [hunk 5]
```diff
@@ -10721,8 +10732,18 @@ <h2 id="AudioWorklet-Examples">
...
-                <code>positionZ</code> AudioParams directly.
+                <code>positionX.value</code>, <code>positionY.value</code>, and
+                <code>positionZ.value</code> attribute directly with the
+                <code>x</code>, <code>y</code> and <code>z</code> parameters,
+                respectively.
+              </p>
+              <p>
+                Consequently, if any of the <code>positionX</code>,
+                <code>positionY</code>, and <code>positionZ</code>
+                <a>AudioParam</a> have an automation curve set using <a href=
```
Comment (rtoy) [inline index.html]: Typo: AudioParam -> AudioParams

File: index.html [hunk 6]
```diff
@@ -10721,8 +10732,18 @@ <h2 id="AudioWorklet-Examples">
...
+                "#dom-audioparam-setvaluecurveattime">setValueCurveAtTime()</a>
```
Comment (rtoy) [inline index.html]: Same link issue as above.

File: index.html (hunk 4 above)
Comment (padenot) [reply index.html]: I've done it.

File: index.html [hunk 7]
```diff
@@ -10696,19 +10697,19 @@ <h2 id="AudioWorklet-Examples">
             </dt>
             <dd>
               <p>
-                This method is DEPRECATED. It is equivalent to setting
-                <code>positionX.value</code>, <code>positionY.value</code>, and
-                <code>positionZ.value</code> attribute directly with the
-                <code>x</code>, <code>y</code> and <code>z</code> parameters,
+                This method is DEPRECATED. It is equivalent to setting <a href=
+                "#dom-pannernode-positionx">positionX.value</a>, <a href=
```
Comment (rtoy) [inline index.html]: I think links should be `<a data-link-for="PannerNode">positionX</a><code>.value</code>`.

File: index.html [hunk 8]
```diff
@@ -11128,20 +11129,22 @@ <h2 id="AudioListener">
...
             <dd>
               <p>
-                This method is DEPRECATED. It is equivalent to setting
-                <code>orientationX.value</code>,
-                <code>orientationY.value</code>,
-                <code>orientationZ.value</code>, <code>upX.value</code>,
-                <code>upY.value</code>, and <code>upZ.value</code> directly
-                with the given <code>x</code>, <code>y</code>, <code>z</code>,
+                This method is DEPRECATED. It is equivalent to setting <a href=
+                "#dom-audiolistener-forwardx">forwardX.value</a>, <a href=
```
Comment (rtoy) [inline index.html]: Similarly, `<a data-link-for="AudioListener">forwardX</a><code>.value</code>`.

File: index.html [hunk 9]
```diff
@@ -11286,20 +11289,20 @@ <h2 id="AudioListener">
             </dt>
             <dd>
               <p>
-                This method is DEPRECATED. It is equivalent to setting
-                <code>positionX.value</code>, <code>positionY.value</code>, and
-                <code>positionZ.value</code> directly with the given
-                <code>x</code>, <code>y</code>, and <code>z</code> values,
-                respectively.
+                This method is DEPRECATED. It is equivalent to setting <a href=
+                "#dom-AudioListener-positionx">positionX.value</a>, <a href=
```
Comment (rtoy) [inline index.html]: Likewise, replace `<a href=...>` with `<a data-link-for="AudioListener">positionX</a><code>.value</code>`

Comment (rtoy) [reply index.html]: I'm ok if you want to land this now.  I'll fix any broken links later.

**Comment by rtoy** [comment]
Merge as is, and I'll fix the link issues.

**Pull request merged**